sdk-core/src/test/java/care/data4life/sdk/RecordServiceCreateRecordTest.kt

```diff
@@ -0,0 +1,624 @@
+
+        Mockito.doReturn(mockEncryptedRecord).`when`(recordService)
+                .encryptRecord(mockDecryptedFhirRecord)
+        Mockito.`when`(mockApiService.createRecord(ALIAS, USER_ID, mockEncryptedRecord))
+                .thenReturn(Single.just(mockEncryptedRecord))
+        Mockito.doReturn(mockDecryptedFhirRecord)
+                .`when`(recordService)
+                .decryptRecord<DomainResource>(
+                        mockEncryptedRecord,
+                        USER_ID
+                )
+        Mockito.doReturn(mockDecryptedFhirRecord)
+                .`when`(recordService)
+                .restoreUploadData(
+                        mockDecryptedFhirRecord,
+                        mockCarePlan,
+                        mockUploadData
+                )
+
+        Mockito.doReturn(mockDecryptedFhirRecord)
+                .`when`(recordService)
+                .uploadData(
+                        mockDecryptedFhirRecord,
+                        null,
+                        USER_ID
+                )
+        Mockito.doReturn(mockMeta).`when`(recordService).buildMeta(mockDecryptedFhirRecord)
+
+        // When
+        val subscriber = recordService.createRecord(mockCarePlan, USER_ID).test().await()
+
+        // Then
+        val record = subscriber
+                .assertNoErrors()
+                .assertComplete()
+                .assertValueCount(1)
+                .values()[0]
+        Truth.assertThat(record.meta).isEqualTo(mockMeta)
+        Truth.assertThat(record.fhirResource).isEqualTo(mockCarePlan)
+        Truth.assertThat(record.annotations).isEqualTo(listOf<String>())
+        inOrder.verify(mockTaggingService).appendDefaultTags(
+                ArgumentMatchers.anyString(),
+                ArgumentMatchers.any<HashMap<String, String>>()
+        )
+        inOrder.verify(mockCryptoService).generateGCKey()
+        inOrder.verify(recordService).uploadData(
+                decryptedRecordIndicator,
+                null,
+                USER_ID
+        )
+        inOrder.verify(recordService).removeUploadData(mockDecryptedFhirRecord)
+
+        inOrder.verify(recordService).encryptRecord(
+                mockDecryptedFhirRecord
+        )
+        inOrder.verify(mockApiService).createRecord(ALIAS, USER_ID, mockEncryptedRecord)
+        inOrder.verify(recordService).decryptRecord<DomainResource>(mockEncryptedRecord, USER_ID)
+        inOrder.verify(recordService).restoreUploadData(
+                mockDecryptedFhirRecord,
+                mockCarePlan,
+                null
+        )
+        inOrder.verify(recordService).buildMeta(mockDecryptedFhirRecord)
+        inOrder.verifyNoMoreInteractions()
+    }
+
+    @Test
+    @Throws(
+            InterruptedException::class,
+            DataRestrictionException.UnsupportedFileType::class,
+            DataRestrictionException.MaxDataSizeViolation::class
+    )
+    fun `Given with unsupported data, createRecord throws an error`() {
```

Comment:
The first suggestion reads much nicer, I have to admit. And just to point something out, which sneaks in otherwise - what was important for me, is to use a active form and make a bold statement/promise about a function. (So active form vs should d smt)

Maybe, once we are in Kotlin for test, we could have some sort of a guide to testing including this name schema?

Reply:
However, I applied schema for this test, but should I rephrase all of them?

Reply:
I rephrased them all...